### PR TITLE
Move two location methods from ap_math to Location

### PR DESCRIPTION
**Pull request by khancyr**
This is the first step to remove AP_Math dependency on location. 
It moves to two methods and correct the header inclusion on some places.
location_update is rename to offset_bearing to be consistant with offset function on Location

### CODE REVIEW

File: APMrover2/mode.cpp [hunk 1]
```diff
@@ -205,7 +205,7 @@ void Mode::set_desired_location(const struct Location& destination, float next_l
     // set final desired speed
     _desired_speed_final = 0.0f;
     if (!is_equal(next_leg_bearing_cd, MODE_NEXT_HEADING_UNKNOWN)) {
-        const float curr_leg_bearing_cd = get_bearing_cd(_origin, _destination);
```
Comment (peterbarker) [inline APMrover2/mode.cpp]: I'd like to clarify this name.  Is it bearing to or from?

`get_bearing_to`, perhaps?

File: libraries/AC_AttitudeControl/AC_PosControl.cpp [hunk 2]
```diff
@@ -2,6 +2,7 @@
 #include "AC_PosControl.h"
 #include <AP_Math/AP_Math.h>
 #include <AP_Logger/AP_Logger.h>
+#include <AP_Common/Location.h>
```
Comment (peterbarker) [inline libraries/AC_AttitudeControl/AC_PosControl.cpp]: Why?

File: libraries/AP_Common/Location.h [hunk 3]
```diff
@@ -85,6 +88,9 @@ class Location
 
     void zero(void);
 
+    // return bearing in centi-degrees between two locations
```
Comment (peterbarker) [inline libraries/AP_Common/Location.h]: Description needs fixing; a bearing is definitely directional :-)

File: libraries/AP_Math/AP_Math.cpp [hunk 4]
```diff
@@ -273,3 +273,18 @@ bool rotation_equal(enum Rotation r1, enum Rotation r2)
     return (v1 - v2).length() < 0.001;
 }
 
+// return horizontal distance between two positions in cm
```
Comment (peterbarker) [inline libraries/AP_Math/AP_Math.cpp]: Should this really be moved?

There may still be room for a location.h

File: libraries/AP_Scripting/lua_bindings.cpp [hunk 5]
```diff
@@ -230,9 +231,7 @@ static int location_project(lua_State *L) {
         return luaL_argerror(L, args, "too many arguments");
     }
 
-    location_update(*check_location(L, -3),
-                    luaL_checknumber(L, -2),
-                    luaL_checknumber(L, -1));
+    *check_location(L, -3).offset_bearing(luaL_checknumber(L, -2), luaL_checknumber(L, -1));
```
Comment (peterbarker) [inline libraries/AP_Scripting/lua_bindings.cpp]: Have you made sure scripting still compiles after this?

Something about `--scripting` to `waf`


We've broken this before...

File: libraries/AP_Math/AP_Math.cpp (hunk 4 above)
Comment (khancyr) [reply libraries/AP_Math/AP_Math.cpp]: I prefer not to confuse everybody by having to location file ...

File: libraries/AC_AttitudeControl/AC_PosControl.cpp (hunk 2 above)
Comment (khancyr) [reply libraries/AC_AttitudeControl/AC_PosControl.cpp]: yep this one as nothing to do here ... I should have paste in the wrong file ... I will double check the other include

**Comment by peterbarker** [comment]
On Fri, 5 Apr 2019, Pierre Kancir wrote:

> +// return horizontal distance between two positions in cm
> 
> I prefer not to confuse everybody by having to location file ...

This isn't really a location-based method we're discussing here, 'though.

If you pass in Vector3fs in metres, I'm pretty sure you won't get 
distances out in cm :-)

I wonder if we could get this into Vector3 somehow? 
"v1.projected_distance(v2)" 
or something?

I'd really prefer to just move the obvious *methods* at this point :-)

**Comment by khancyr** [comment]
I have remove the part that move functions not related to Location

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**